### DISCUSSION TITLE
Some way to determine the type of a component in the tree

### DISCUSSION THREAD

**Discussion by dncpax** [Budibase/budibase#2648] [Ideas]
I may be missing the obvious, but I'm trying to replicate a "details/editing" screen created by bb, and cannot see what kind of object each item is. For instance, what's the "breadcrumbs" component and subcomponents?

Thanks.

**Comment by shogunpurple** [comment]
@dncpax The breadcrumbs component is a predefined component that comes with our autoscreens. Underneath, it's just a container component that we use to render the links inside it separated by arrows.

**Reply by dncpax** [reply]
breadcrumbs was just an example. when replicating a page you have to wonder what each component really is...

**Reply by mjashanks** [reply]
I get what you're saying, because....

![image](https://user-images.githubusercontent.com/3524181/134002542-06097b19-fa29-4225-9133-8a0108377ad8.png)

Does not tell you what the components actually are. Even if you click on a component, nowhere does it say "Container" etc.

This is a good point, as people will like to learn Budibase by reverse-engineering premade screens or templates.

I've renamed this discussion, and set it to an "Idea".
